txjuju/cli.py

```diff
@@ -13,9 +14,124 @@
 from twisted.internet.protocol import ProcessProtocol
 from twisted.python import log
 
+from . import config, _utils
 from .errors import CLIError
 
 
+def get_executable(filename, version_cli, cfgdir, envvars=None):
+    """Return the Executable for the given juju binary.
+
+    @param filename: The path to the juju binary.
+    @param version_cli: The version-specific JujuXCLI to use.
+    @param cfgdir: The Juju config dir to use.
+    @param envvars: The additional environment variables to use.
+    """
+    if not version_cli:
```

Comment:
Is this needed needed? can't we just rely on the mandatory positional parameters to catch this and cfgdir?